### DISCUSSION TITLE
V4 Dev Server without reinstalling dependencies

### DISCUSSION THREAD

**Discussion by gramian** [lowdefy/lowdefy#1797] [Questions and Help] [answered]
Dear All,
I am trying to build a Docker container which starts a dev server via a `pnpx lowdefy@4 dev --watch /mymount` command.
However everytime I start this container, the dependencies of the separate `@lowdefy/server-dev` module are reinstalled, which takes too long.
I tried already to install `@lowdefy/server-dev` beforehand and point `lowdefy` to it via the `--dev-directory` argument. And I also I tried to add `pnpm install` in the dev server folder. Neither prevented the installation of dependencies when the container is started.

For example: What do I need to add here?
```dockerfile
RUN pnpm add lowdefy@4

CMD pnpx lowdefy@4 dev --watch /mymount
```

Thank you for your help

**Comment by Gervwyk** [comment]
Hi @gramian
The dev server handles a bunch of logic to make developing lowdefy easier. Are you running the dev server in "prod" in order to make use of the `--watch` functionality?

**Reply by gramian** [reply]
No,
but I have a dev environment based on containers.

If I understand you correctly, you say the `dev` command is justfor developing on the lowdefy package itself and I should use `lowdefy start` with the `watch:` object for developing a page.

**Reply by gramian** [reply]
But wouldn't this require a rebuild every time I change a source `.yaml` file?

**Comment by gramian** [comment, marked as answer]
I found a solution:

```dockerfile

RUN pnpm add lowdefy@4

RUN pnpm add @lowdefy/server-dev@4

RUN mkdir -p .lowdefy/dev

RUN mv node_modules/@lowdefy/server-dev/* .lowdefy/dev

RUN cd .lowdefy/dev && pnpm install --no-frozen-lockfile

CMD pnpm exec lowdefy dev --watch /mymount
```

This is not very elegant, but simpler variants did not work.
